big_tests/tests/mam_SUITE.erl

```diff
@@ -2974,6 +2975,19 @@ check_user_exist(Config) ->
   %% cleanup
   ok = escalus_ejabberd:rpc(ejabberd_auth, remove_user, [AdminU, AdminS]).
 
+decode_compact_uuid(_Config) ->
+    %% ensure we can handle time based UUIDs
+    quickrand:seed(),
+    UState = uuid:new(self()),
+    {Uuid, _NewState} = uuid:get_v1(UState),
+
+    %% would crash if not...
+    mod_mam_utils:decode_compact_uuid(list_to_binary(uuid:uuid_to_string(Uuid))),
```

Comment:
Big tests don't have access to mongoose modules (or at least we don't do calls to mongoose modules like this).
Still, an RPC should be possible.

Reply:
To enable the test, it should be added into 

```erlang
impl_specific() ->                                                                   
  [check_user_exist,
   decode_compact_uuid].
```

A good test would also try to query archive, using compact uuids.

Reply:
Yea, I'm aware it's a bit weird what I'm doing. That's why I asked for eunit tests, since I think that would be the correct approach to show this kind of functionality works as expected.

For a true system test I'd have to ship a custom backend module that would handle external UUIDs. What would be the right place to have this? Also I'd have to change the configuration of the node under test to use that custom backend. Any pointers how to do that?

Reply:
ok, as Piotr said, small tests are in test/ directory.
I am not sure if we need a system wide test here.

But if I would need to write one, that the module would go to src/ OR to big_tests/tests/ and dynamically loaded (code:load_binary). Than mam_SUITE can be used to test the backend. mam_SUITE already tests different configurations of MAM (for example, async writers).